src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java

```diff
@@ -551,11 +662,11 @@ private Object readResolve() {
                 storageAccountNameReferenceType, newStorageAccountName, existingStorageAccountName);
 
         if (StringUtils.isBlank(imageTopLevelType)) {
-            if (StringUtils.isNotBlank(image)
-                    || StringUtils.isNotBlank(imageId)
-                    || StringUtils.isNotBlank(imageOffer)
-                    || StringUtils.isNotBlank(imageSku)
-                    || StringUtils.isNotBlank(imagePublisher)) {
+            if (imageReference != null  && (StringUtils.isNotBlank(imageReference.getUri())
```

Comment:
This contains an obscure bug here. The `imageReference` variable will be null all the time.